src/ser.rs

```diff
@@ -128,7 +129,7 @@ impl<'a> ser::Serializer for &'a Serializer {
     }
 
     fn serialize_f64(self, v: f64) -> Result<Term> {
-        Ok(Term::Float(eetf::Float::from(v)))
+        Ok(Term::Float(eetf::Float::try_from(v)?))
```

Comment:
Nice. I don't pay too close attention to Rust so didn't realise TryFrom had finally been stabilised.